libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -234,12 +234,94 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 
 void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
 {
-	if (type(_unaryOperation).category() == Type::Category::RationalNumber)
+	Type const& resultType = type(_unaryOperation);
+	Token const op = _unaryOperation.getOperator();
+
+	if (resultType.category() == Type::Category::RationalNumber)
+	{
 		defineExpression(_unaryOperation) <<
-			formatNumber(type(_unaryOperation).literalValue(nullptr)) <<
+			formatNumber(resultType.literalValue(nullptr)) <<
 			"\n";
+	}
+	else if (resultType.category() == Type::Category::Integer)
+	{
+		solAssert(resultType == type(_unaryOperation.subExpression()), "Result type doesn't match!");
+
+		if (op == Token::Inc || op == Token::Dec)
+		{
+			solAssert(!!m_currentLValue, "LValue not retrieved.");
+			string fetchValueExpr = m_currentLValue->retrieveValue();
+			string modifiedValue = m_context.newYulVariable();
+			string originalValue = m_context.newYulVariable();
+
+			m_code << "let " << originalValue << " := " << fetchValueExpr << "\n";
+			m_code <<
+				"let " <<
+				modifiedValue <<
+				" := " <<
+				(op == Token::Inc ?
+					m_utils.incrementCheckedFunction(resultType) :
+					m_utils.decrementCheckedFunction(resultType)
+				) <<
+				"(" <<
+				originalValue <<
+				")\n" <<
+				m_currentLValue->storeValue(modifiedValue, resultType);
+
+			defineExpression(_unaryOperation) <<
+				(_unaryOperation.isPrefixOperation() ? modifiedValue : originalValue) <<
+				"\n";
+
+			m_code << m_currentLValue->storeValue(modifiedValue, resultType);
+			m_currentLValue.reset();
+		}
+		else if (op == Token::BitNot)
+			appendSimpleUnaryOperation(_unaryOperation, _unaryOperation.subExpression());
+		else if (op == Token::Add)
+		{
+			// Nothing to do
```

Comment:
You still need to call `defineExpression`, otherwise code like `uint a = +x;` will translate to `let _1 := x; a := _2; ` with `_2` representing `+x` but being undefined.

Reply:
Understood